src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +239,199 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
-                    notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_RESULT);
-                } catch (JsonProcessingException e) {
-                    LOGGER.error(e.toString());
-                }
+    public void consumeCalculationStopped(Message<String> msg, ComputationType computationType) {
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        if (!Strings.isBlank(receiver)) {
+            NodeReceiver receiverObj;
+            try {
+                receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                // delete computation results from the database
+                networkModificationTreeService.updateComputationResultUuid(receiverObj.getNodeUuid(), null, computationType);
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+                // send notification for stopped computation
+                notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), computationType.getUpdateStatusType());
+
+                LOGGER.info("{} stopped for node '{}'", computationType.getLabel(), receiverObj.getNodeUuid());
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.toString());
             }
-        };
+        }
     }
 
-    @Bean
-    public Consumer<Message<String>> consumeSensitivityAnalysisStopped() {
-        return message -> {
-            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
-            if (receiver != null) {
-                NodeReceiver receiverObj;
-                try {
-                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+    public void consumeCalculationResult(Message<String> msg, ComputationType computationType) {
+        UUID resultUuid = null;
+        String resultId = msg.getHeaders().get(RESULT_UUID, String.class);
+        if (resultId != null) {
+            resultUuid = UUID.fromString(resultId);
+        }
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        if (!Strings.isBlank(receiver)) {
+            NodeReceiver receiverObj;
+            try {
+                receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                ComputationType updatedComputationType = computationType;
+                if (updatedComputationType == SHORT_CIRCUIT) {
+                    String busId = msg.getHeaders().get(HEADER_BUS_ID, String.class);
+                    if (!StringUtils.isEmpty(busId)) {
+                        updatedComputationType = SHORT_CIRCUIT_ONE_BUS;
+                    }
+                }
```

Comment:
same and move it in consumeShortCircuitAnalysisResult

Reply:
Very good idea. Done.